### ISSUE TITLE
[BUG] Maximum call stack size exceeded when using the same plugin in 2 different plugins

### ISSUE THREAD

**Issue by Luisetepe** [elysiajs/elysia#150]
Hi! I have defined 2 Elysia instances (plugins) in different files to conceptually represent different resource routes, but they both use a 3rd plugin that declares my store.
`api.dependencies.ts`
```typescript
export const apiSetup = new Elysia({ name: 'apiSetup' }).state('useCases', {
    getAllProductsUseCase,
    getAllProductCategoriesUseCase,
    createProductUseCase,
    createProductCategoryUseCase
})
```
`products.controller.ts`
```typescript
export const apiProducts = new Elysia({ prefix: '/products' })
    .use(apiSetup)
   /*Implementation here */
```
`product-categories.controller.ts`
```typescript
export const apiProductCategories = new Elysia({ prefix: '/product-categories' })
    .use(apiSetup)
   /*Implementation here */
```
But this setup gives the following error:
![image](https://github.com/elysiajs/elysia/assets/3601697/5611832d-99b3-4642-a7ea-637509eb2e1a)

If I put the `.use(apiSetup)` only in my `index.ts` in my root Elysia instance, everything works OK, but obviously I dont get type safete or intellisense in my routes.
Per the documentation, I understood that from v0.6, you could put a {name} in your plugins in order for them to automatically deducplicate themselves. 
I'm I doing something wrong?

**Comment by Gsiniguez** [comment]
I have the same issua when i used prisma client in multiples routes by decorator. I export prisma instance and used like that and not in dependency injection. Some libraries not suppor multiple calls i guess

**Comment by SaltyAom** [comment]
Hi, we are trying to tackle this problem.

Can you help me by providing reproducible code would be a lot of help. Thanks!

**Comment by Luisetepe** [comment]
@SaltyAom sure! Sorry for the delay, different timezone I think 😆 
I created this repo using SQLite (already has a Database.db in it) [](https://github.com/Luisetepe/elysiajs-issue-150)
bun install && bun run dev should give the error. If you remove the `.use(apiSetup)` calls and put it in the root `index.ts` then it works perfectly fine.

Tell me if I can help with anything else!

**Comment by bogeychan** [comment]
Hi all 👋, the mentioned repo: https://github.com/Luisetepe/elysiajs-issue-150

I was able to reproduce it:

```bash
RangeError: Maximum call stack size exceeded.
      at ./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238557
      at forEach (:1:20)
      at n0 (./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238390)
      at ./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238557
      at forEach (:1:20)
      at n0 (./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238390)
      at ./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238557
      at forEach (:1:20)
      at n0 (./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238390)
      at ./elysiajs-issue-150/node_modules/elysia/dist/bun/index.js:2:238557
```

**Comment by Luisetepe** [comment]
Sorry! I didn't know the link would not render in the comment If I didn't give it some text 😞 . I've updated the comment anyways. Thanks @bogeychan !

**Comment by Luisetepe** [comment]
@SaltyAom  and @bogeychan just an update. I updated to 0.6.23 cause I saw that the merge was about fixing the problem, tried it out and it does "resolve" the problem, but it messes up with the `.state()' function that injects dependecies to the store (I've tried to rollback to 0.6.21, and it fixes the store error). For example I'm injecting this class to the `.state()`:
```typescript
export class GetAllProductCategoriesUseCase {
    private readonly _repo: IProductCategoryRepository

    constructor(repo: IProductCategoryRepository) {
        this._repo = repo
    }

    Execute(pagination?: PaginationOptions) {
        return this._repo.getAllProductCategories(pagination)
    }
}
```
and then I do:
```typescript
const productCategoryRepository: IProductCategoryRepository = new ProductCategoryRepository(dbPool)
const getAllProductCategoriesUseCase = new GetAllProductCategoriesUseCase(productCategoryRepository)

// Inject all dependencies
export const apiSetup = new Elysia({ name: 'apiSetup' }).state('useCases', {
    getAllProductCategoriesUseCase
})
```
If I access the store in a handler:
```typescript
.get(
        '/',
        async ({ store, query }) => {
            const pagination = { page: query.page ?? 1, pageSize: query.pageSize ?? 10 }
            const categories =
                await store.useCases.getAllProductCategoriesUseCase.Execute(pagination)
```
It gives me an error that the object `store.useCases.getAllProductCategoriesUseCase`does not contain the "Execute()" function.
If I console log what it is, this is what I get:
![image](https://github.com/elysiajs/elysia/assets/3601697/65b7648b-5211-4047-8eae-e38197a3aa3b)

So it is getting that is an object that has a private property in it `_repo` which itself has a reference to the Drizzle ORM (that were all the things in the log come from), but the function Execute() does not appear anywhere. Very strange that is removing it in some way

**Comment by SaltyAom** [comment]
Should be fixed in 0.7.0, will await for your confirmation.

**Comment by Luisetepe** [comment]
Hi @SaltyAom. With my repo [https://github.com/Luisetepe/elysiajs-issue-150](https://github.com/Luisetepe/elysiajs-issue-150), if you git clone and run `bun install elysia@0.7.1` and `bun run dev`, I got this:

<img width="1206" alt="image" src="https://github.com/elysiajs/elysia/assets/3601697/43e2240f-e034-4e04-8b4b-90abe51ec41f">

<img width="1209" alt="image" src="https://github.com/elysiajs/elysia/assets/3601697/6c1afbbc-11ca-424b-a22b-3fc50f51d4ba">

**Comment by SaltyAom** [comment]
Should be fixed with >= 0.7.10 with dedicated `deepMerge` for custom class and FFI.

Feels free to reopen if the issue still exists.
